src/platform/android/java/chip/platform/NsdManagerServiceResolver.java

```diff
@@ -127,9 +127,15 @@ public void publish(
       String[] subTypes) {
     NsdServiceInfo serviceInfo = new NsdServiceInfo();
     serviceInfo.setServiceName(serviceName);
-    serviceInfo.setServiceType(type);
+
+    StringBuilder sb = new StringBuilder(type);
+    for (String subType : subTypes) {
+      sb.append(",").append(subType);
```

Comment:
This looks like some custom representation. How does this get used? Please document the assumption s

Reply:
In fact, this is not a custom representation, Android mdns is using apple mDNSResponder, which supports this format to publish subtype of a service. I have tested this on some android phones without any changes.

But yes, I also do not find any document about this, I searched this way here https://stackoverflow.com/questions/9694613/subtype-and-mdnsresponder

Reply:
The fact that some versions of android may be implementing the API with mDNSResponder, and that the given version of mDNSResponder code (which could change over time) allows this, is merely a reverse engineering of one instance. The `NsdServiceInfo.setServiceType` API is a portable Android framework API. There is no behavioral guarantee on this workaround which actually just employs the fact found that an internal implementation detail does insufficient validation. The API says `setServiceType` (not plural) and subtypes are mentioned nowhere in NsdServiceInfo (https://developer.android.com/reference/android/net/nsd/NsdServiceInfo). 

The approach taken here is not future-proof, and may not work with some Android integrations that have replaced the backend implementation from those where you attempted.

I would be OK to support this change with the following specific updates:

- [ ] Mention that subtypes registration is an *undocumented* feature that was discovered to work the way done here by experimentation
- [ ] Mention that this method MAY STOP WORKING in future Android versions since a standard API is used in a non-standard way
- [ ] More clearly explain what is being done (and the fact it relies on mDNSResponder backend implementation behavior) and how it was tested.

Reply:
> In fact, this is not a custom representation, Android mdns is using apple mDNSResponder, which supports this format to publish subtype of a service. I have tested this on some android phones without any changes.
> 
This is only an implementation detail of one implementation of Android. The public API itself of android does not guarantee this support.

Reply:
Do you mean that I should add comment about this change in my PR?

Reply:
as discuss in slack, comment is added